lib/puppet/functions/join.rb

```diff
@@ -0,0 +1,99 @@
+# Joins an array of values into a string with elements separated by a delimiter.
+#
+# Supports up to three arguments
+# * first argument is required and must be an `Array`, or an `Iterable`
```

Comment:
I find this confusing given that an `Array` in fact is an `Iterable`. I think it would be better to either mention just `Iterable` or mention all types that are `Iterable`.

Reply:
ok, will modify (will need to leave a list of Iterable open ended though as we may add more)